.github/workflows/ngsa.yaml

```diff
@@ -183,16 +181,21 @@ jobs:
     - name: Docker Tag and Push
       if: ${{ env.DOCKER_SET == 'true' }}
       run: |
-        # tag the repo with :beta
+        if [[ -z "${APP_VERSION}" ]]; then
+          echo "APP_VERSION not set, cannot proceed with docker tag and push"
+          exit 1
+        fi
+        
+        # Tag the image with :beta and semver
+        semver=$APP_VERSION-${{ github.run_number }}
         docker tag image $DOCKER_REPO:beta
+        docker tag image $DOCKER_REPO:beta-$semver
```

Comment:
one thing we need to think about is how we cleanup images. We're going to have a LOT of images with this new approach. can you put an issue on the board to address?